### PR TITLE
feat: TC-48 add edit discount page

### PR DESCRIPTION
**Pull request by CristiPV**
### This pull request closes:

🏁 Jira Issue: [TC-48](https://treecreate.atlassian.net/jira/software/c/projects/TC/boards/1?modal=detail&selectedIssue=TC-48)

### Which service(s) does this issue affect:

- [ ] 🛰️ API
- [ ] 🛒 webstore
- [x] 🎓 admin-page

### Types of changes

- [ ] 🐛 Bug fix (a non-breaking change which fixes an issue)
- [x] 🧱 New feature (a non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🤖 CI/CD (a change to the CI/CD pipeline)
- [ ] ⚗️ Refactoring (an update to some already existing code)

### Documentation Updates

- [x] 📂 No need for updates
- [ ] 📁 Requires an update ( _not done within this issue_ )
- [ ] 🗃️ Has been updated ( _done within this issue_ )

### Testing checklist

- [x] 💪 Manual tests
- [x] 🔧 Automatic tests

#### Browser compatibility - _**[**Frontend only**]**_

- [ ] Tested on Safari
- [x] Tested on Chrome
- [x] Tested on Mozilla Firefox

### What changes have been done within this issue?

I created the edit discount page. It can be accessed via the 'edit' button beside any discount on the discounts page and should display the relevant information for each discount.

### How should this be manually tested?

By going to the edit page of any discount and checking if the information is correct.

### CODE REVIEW

File: apps/admin-page/src/app/pages/edit-discount/edit-discount.component.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    <mat-icon data-cy="icon">arrow_back_ios</mat-icon> Back
+</button>
+<webstore-item-card
+    title="Edit discount"
+    [content]="content"
+    [header]="header"
+    [isLoading]="isLoading"
+    data-cy="edit-discount-card"
+    class="edit-discount-content"
+>
```
Comment (Teodor25) [inline apps/admin-page/src/app/pages/edit-discount/edit-discount.component.html:17]: Should be a form. So add a <form [formControl]="editDiscountForm" ....> {{ add everything here and change input fields from [formControll]="..." => formControlName="..." }} </form> for an example, check account page. 

Remember to add error messages as well. For examples look at account page.

File: apps/admin-page/src/app/pages/edit-discount/edit-discount.component.html [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                        <mat-label>Discount code</mat-label>
+                        <input matInput type="text" [formControl]="discountCodeControl" data-cy="discount-code" />
+                    </mat-form-field>
+                    <div class="spacer"></div>
+                    <mat-form-field appearance="fill">
+                        <mat-label>Uses left</mat-label>
+                        <input matInput type="number" [formControl]="usesLeftControl" data-cy="discount-uses-left" />
+                    </mat-form-field>
+                    <div class="spacer"></div>
+                    <mat-form-field appearance="fill">
```
Comment (Teodor25) [inline apps/admin-page/src/app/pages/edit-discount/edit-discount.component.html]: Make this into a date picker. Same with start date.

File: apps/admin-page/src/app/pages/edit-discount/edit-discount.component.html [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                        />
+                    </mat-form-field>
+                </div>
+                <div class="row">
+                    <mat-form-field appearance="fill">
+                        <mat-label>Amount</mat-label>
+                        <input matInput type="number" [formControl]="amountControl" data-cy="discount-amount" />
+                    </mat-form-field>
+                    <div class="spacer"></div>
+                    <mat-form-field appearance="fill">
```
Comment (Teodor25) [inline apps/admin-page/src/app/pages/edit-discount/edit-discount.component.html]: This should be a select. value="AMOUNT" && value="PERCENT"

File: apps/admin-page/src/app/pages/edit-discount/edit-discount.component.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+   * \
+   * **Uses**: `discountService` -> to call the API.
+   */
+  fetchDiscount(id: string): void {
+    this.isLoading = true;
+    this.discountService.getDiscounts().subscribe({
+      error: (error: HttpErrorResponse) => {
+        console.error(error);
+      },
+      next: (discounts: IDiscount[]) => {
```
Comment (Teodor25) [inline apps/admin-page/src/app/pages/edit-discount/edit-discount.component.ts]: When possible, we need to add a findById to discountService.

**Review by Teodor25 — CHANGES_REQUESTED** [review]
implement commented changes from past review.

File: apps/admin-page/src/app/pages/edit-discount/edit-discount.component.ts (hunk 4 above)
Comment (Kwandes) [reply apps/admin-page/src/app/pages/edit-discount/edit-discount.component.ts]: Already added to the API in #284 so it can be integrated right away

**Review by Teodor25 — APPROVED** [review]

**Pull request merged**